packages/react-sdk/src/components/NoiseCancellation/NoiseCancellationProvider.tsx

```diff
@@ -76,10 +76,31 @@ export const NoiseCancellationProvider = (
   const hasCapability = useHasPermissions(
     OwnCapability.ENABLE_NOISE_CANCELLATION,
   );
+  const [isSupportedByBrowser, setIsSupportedByBrowser] = useState<
+    boolean | undefined
+  >(undefined);
+
+  useEffect(() => {
+    const result = noiseCancellation.isSupported();
+
+    if (typeof result === 'boolean') {
+      setIsSupportedByBrowser(result);
+    } else {
+      result
+        .then((_isSupportedByBrowser) =>
+          setIsSupportedByBrowser(_isSupportedByBrowser),
+        )
+        .catch((err) =>
+          console.error(
+            `Can't determine if noise cancellation is supported`,
+            err,
+          ),
+        );
```

Comment:
```suggestion
      let cancel = false;
      result
        .then((_isSupportedByBrowser) => {
          if (!cancel) {
            setIsSupportedByBrowser(_isSupportedByBrowser)
          }
        })
        .catch((err) =>
          console.error(
            `Can't determine if noise cancellation is supported`,
            err,
          ),
        );
        return () => (cancel = true);
```

Reply:
Also, can we add a quick note somewhere in the doc comments what "undefined" means for `isSupported`?

Reply:
Yes, updated the react guide